pysat/_meta.py

```diff
@@ -272,7 +272,7 @@ def drop(self, names):
     def keep(self, keep_names):
         """Keeps variables (keep_names) while dropping other parameters"""
 
-        current_names = self._data.columns
+        current_names = self._data.index
```

Comment:
What instigated this change?

Reply:
`keep` and `drop` should be similar functions.  I believe that both should act on the index in order to keep or drop metadata names (`keep` actually calls `drop`, so an inconsistency results in a KeyError).

test code:
```
import pysat
test = pysat.Instrument('pysat', 'testing')     
test.load(date=pysat.datetime(2009,1,1))  
test.meta.data  
```
yields
```
                                     units            long_name  ... custom Bin_Location
uts                                      s       Universal Time  ...  False          NaN
Epoch          Milliseconds since 1970-1-1                Epoch  ...    NaN          0.5
mlt                                  hours  Magnetic Local Time  ...    NaN          NaN
slt                                  hours     Solar Local Time  ...    NaN          NaN
orbit_num                                          Orbit Number  ...    NaN          NaN
longitude                          degrees            Longitude  ...    NaN          NaN
latitude                           degrees             Latitude  ...    NaN          NaN
dummy1                                                   dummy1  ...    NaN          NaN
dummy2                                                   dummy2  ...    NaN          NaN
dummy3                                                   dummy3  ...    NaN          NaN
dummy4                                                   dummy4  ...    NaN          NaN
string_dummy                                       string_dummy  ...    NaN          NaN
unicode_dummy                                     unicode_dummy  ...    NaN          NaN
int8_dummy                                           int8_dummy  ...    NaN          NaN
int16_dummy                                         int16_dummy  ...    NaN          NaN
int32_dummy                                         int32_dummy  ...    NaN          NaN
int64_dummy                                         int64_dummy  ...    NaN          NaN
```
Calling 
```
test.meta.drop('int64_dummy')    
```
yields
```
                                     units            long_name  ... custom Bin_Location
uts                                      s       Universal Time  ...  False          NaN
Epoch          Milliseconds since 1970-1-1                Epoch  ...    NaN          0.5
mlt                                  hours  Magnetic Local Time  ...    NaN          NaN
slt                                  hours     Solar Local Time  ...    NaN          NaN
orbit_num                                          Orbit Number  ...    NaN          NaN
longitude                          degrees            Longitude  ...    NaN          NaN
latitude                           degrees             Latitude  ...    NaN          NaN
dummy1                                                   dummy1  ...    NaN          NaN
dummy2                                                   dummy2  ...    NaN          NaN
dummy3                                                   dummy3  ...    NaN          NaN
dummy4                                                   dummy4  ...    NaN          NaN
string_dummy                                       string_dummy  ...    NaN          NaN
unicode_dummy                                     unicode_dummy  ...    NaN          NaN
int8_dummy                                           int8_dummy  ...    NaN          NaN
int16_dummy                                         int16_dummy  ...    NaN          NaN
int32_dummy                                         int32_dummy  ...    NaN          NaN
```

Calling 
```
test.meta.keep('uts')
```
should yield
```
                                     units            long_name  ... custom Bin_Location
uts                                      s       Universal Time  ...  False          NaN
```
In the current develop branch, this results in a KeyError, since 'uts' does not appear in the list of columns.

I could be misreading the code, and drop should be the one to be changed so that both operate on the columsn.  @rstoneback?

Reply:
You have it correct, the keep should drop of all of the other variable names, stored in the index.